src/Runner.Common/RunServer.cs

```diff
@@ -70,7 +70,43 @@ private void CheckConnection()
         public Task<AgentJobRequestMessage> GetJobMessageAsync(string id)
         {
             CheckConnection();
-            return _taskAgentClient.GetJobMessageAsync(id);
+            var jobMessage = RetryRequestWithTimeout<AgentJobRequestMessage>(async () =>
```

Comment:
plan
- will leave the retries in this file
- pass the cancellation token (in case runner is shutting down; `messageQueueLoopTokenSource.Token` from runner.cs)
- let's switch to count-based retries (5 retries; keep the 5-15s backoff). Argument is purely about consistency. Same algorithm for reader bouncing between files. Apologies, this is my fault for the original time-based proposal :(
- will not move to a retry util for now, since would require caller to provide retriable. Today we're using taskAgentClient but tomorrow switching to ___.

Reply:
I have reworked retries to  count-based, and started to pass cancellation token